background/basicauth.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * @copyright Robin Jadoul
+ *
+ * Copyright (C) 2018  Robin Jadoul
+ * This file is part of Keywi.
+ * Keywi is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Keywi is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Keywi.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+(function() {
+  function extractHost(url) {
+    // Extract host from '<protocol>://<host>/<rest>', includes the port
+    return (/:\/\/([^/]+)\//).exec(url)[1];
+  }
+
+  function  confirmationDialog(config) {
+    const dialogUrl = browser.extension.getURL('dialog/confirm_basic_auth.html');
+    return browser.windows.create({
+      'type': 'panel',
+      'width': 400,
+      'height': 600,
+      'url': dialogUrl,
+      'incognito': true
+    }).then(function (newWindow) {
+      const openedWindowId = newWindow.id;
+
+      return new Promise(function(resolve, reject) {
+        browser.tabs.query({"windowId": openedWindowId}).then(function (tabs) {
+          const openTabId = tabs[0].id;
+          browser.tabs.onUpdated.addListener(function _sendData(tabId, changeInfo, tabInfo) {
+            if (tabId === openTabId && tabInfo.status === 'complete') {
+              setTimeout(function () {
+                browser.tabs.sendMessage(openTabId, {
+                  'type': 'confirm_basic_auth_data',
+                  'data': config
+                });
+                browser.tabs.onUpdated.removeListener(_sendData);
+              }, 300);
+            }
+          });
+        });
+
+        function _onRemove(winId) {
+          if (openedWindowId === winId) {
+            clean();
```

Comment:
Maybe we should add `// eslint-disable-line no-use-before-define`
(also lines 65, 71, 74)